ql/src/java/org/apache/hadoop/hive/ql/io/HdfsUtils.java

```diff
@@ -117,6 +117,10 @@ public static long createTestFileId(
     }
     return result;
   }
+  public static List<Path> listPath(final FileSystem fs, final Path path, final PathFilter filter,
```

Comment:
1. if you could process the results with any computation per record, you will get full benefits of the async page fetch offered by s3a and (soon) abfs; at 600ms a list for 200 records on s3, that's potentially 3ms/record saving
2. If `listLocatedFileStatus()` logged @ debug the toString() value of the iterator, the S3A FS iterator will print out its IOStats, including #of S3 list requests and  min/mean/max durations.

Reply:
Thanks @steveloughran for taking a look at this. I will try to take advantage of the async prefetch feature in later PRs, it seems promising, but it would need bigger code change. I will checkout the IOStats downstream, I have seen it is already available there.